lib/recipe/widgets/ingredient_screen.dart

```diff
@@ -0,0 +1,272 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:gibsonify/recipe/recipe.dart';
+import 'package:dropdown_plus/dropdown_plus.dart';
+
+class IngredientForm extends StatelessWidget {
+  final int recipeIndex;
+  final int ingredientIndex;
+  IngredientForm(this.recipeIndex, this.ingredientIndex, {Key? key})
+      : super(key: key);
+
+  final List<String> cookingStates = [
+    "Raw",
+    "Boiled",
+    "Boiled in water but retained water",
+    "Boiled in water but removed water",
+    "Steamed",
+    "Roasted with oil",
+    "Roasted without oil",
+    "Fried",
+    "Stir - fried",
+    "Soaking and stir fried",
+    "Boiled and fried",
+    "Boiled and stir-fried",
+    "Steamed and fried",
+    "Roasted and boiled"
+  ];
+  final List<String> measurementMethods = [
+    "Direct weight",
+    "Volume of water",
+    "Volume of food",
+    "Play dough",
+    "Number",
+    "Size (photo)"
+  ];
+  final List<String> sizes = [
+    "Small Spoon",
+    "Big spoon",
+    "Standard cup",
+    "Small",
+    "Medium",
+    "Large"
+  ];
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<RecipeBloc, RecipeState>(builder: (context, state) {
+      return Padding(
+          padding: const EdgeInsets.all(8.0),
+          child: Column(
+            children: [
+              TextFormField(
+                initialValue: state.recipes[recipeIndex]
+                    .ingredients[ingredientIndex].name.value,
+                decoration: InputDecoration(
+                  icon: const Icon(Icons.set_meal_rounded),
+                  labelText: 'Ingredient name',
+                  helperText: 'Ingredient name e.g. Rice',
+                  errorText: state.recipes[recipeIndex]
+                          .ingredients[ingredientIndex].name.invalid
+                      ? 'Enter an ingredient name e.g. Tomato'
+                      : null,
+                ),
+                onChanged: (value) {
+                  context.read<RecipeBloc>().add(IngredientNameChanged(
+                      ingredient: state
+                          .recipes[recipeIndex].ingredients[ingredientIndex],
+                      ingredientName: value,
+                      recipe: state.recipes[recipeIndex]));
+                },
```

Comment:
None of the actual FormFields on this page are actually emitting the Events that they should. This is the main bug that needs to be fixed for the widgets to be rebuilding.

See a log of my BlocObserver printing the transitions of the RecipeBloc along with the related Events and States. Nothing is happening when interacting with anything on the `ingredient_page`:

https://user-images.githubusercontent.com/45038919/141223876-abc17a0c-0367-42be-87f4-a1069ee0cac5.mp4